relay/nakama/custom_auth.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+
+	"github.com/heroiclabs/nakama-common/api"
+	"github.com/heroiclabs/nakama-common/runtime"
+	"github.com/rotisserie/eris"
+	"google.golang.org/grpc/codes"
+
+	"pkg.world.dev/world-engine/relay/nakama/siwe"
+	"pkg.world.dev/world-engine/relay/nakama/utils"
+)
+
+const (
+	signInWithEthereumType = "siwe"
+)
+
+var (
+	ErrBadCustomAuthType = errors.New("bad custom auth type")
+)
+
+func initCustomAuthentication(initializer runtime.Initializer) error {
+	if err := initializer.RegisterBeforeAuthenticateCustom(handleCustomAuthentication); err != nil {
+		return eris.Wrap(err, "failed to init siwe authentication")
```

Comment:
nit: while this is true for now, when we start adding more custom authentication this error message is no longer valid. It's better for this to be `failed to register custom authentication handlers` the `failed to init siwe authentication` can then be bubbled up from the `if authType == signInWithEthereumType` block by wrapping the error from handler.

Reply:
I updated the comment to not include the siwe part, however we don't have the option of emitting a separate error for each custom authentication type. The RegisterBeforeAuthenticateCustom only supports registering a single function, so as we add on more custom authentication types, they will all have to go into the same handler.